### PR TITLE
Add Refresh action and toolbar for knative tree

### PR DESCRIPTION
**Pull request by evidolob**
Fix: #13 

Demo:
Screenshot 2021-03-11 at 15 28 52

### CODE REVIEW

**Comment by lstocchi** [comment]
The refresh button to the top is quite problematic for me. It can also be useful in some context but the behavior is not what i would expect. 
As this is in the top toolbar and external to the tree i would expect that all the tree is refreshed when clicked but the focus is on the node selected so it's like i execute refresh on that node.
The other issue is the visibility, when opening some tab (e.g TODO tab) the toolbar disappear. 
refresh

**Comment by evidolob** [comment]
@lstocchi yeh, I can modify behaviour of refresh action, it will be depends on context on which action executed
And I just disable action on toolbar instead of hiding it.

**Comment by evidolob** [comment]
@lstocchi I have change behaviour of refresh action, if it executed from toolbar it will refresh all tree, also I disable action on toolbar instead of hiding:

https://user-images.githubusercontent.com/929743/110913749-2e102680-831e-11eb-91e8-00340e7be299.mov

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -25,6 +25,12 @@
     <actions>
         <group id="com.redhat.devtools.intellij.knative.tree" popup="true">
             <action class="com.redhat.devtools.intellij.knative.actions.OpenInEditorAction" text="Open in Editor"/>
+            <action id="Knative.Refresh" class="com.redhat.devtools.intellij.knative.actions.RefreshAction"
+                    text="Refresh" icon="AllIcons.Actions.Refresh"/>
+        </group>
+
+        <group id="Knative.View.ActionsToolbar">
```
Comment (lstocchi) [inline src/main/resources/META-INF/plugin.xml:32]: i think it is better to add `com.redhat.devtools.intellij` here

File: src/main/java/com/redhat/devtools/intellij/knative/Constants.java [hunk 2]
```diff
@@ -12,4 +12,5 @@
 
 public class Constants {
     public static final String STRUCTURE_PROPERTY = Constants.class.getPackage().getName() + ".structure";
+    public static final String TOOLBAR_PLACE = "Knative.View";
```
Comment (lstocchi) [inline src/main/java/com/redhat/devtools/intellij/knative/Constants.java:15]: better to add the package name here as the structure so if there will ever be another plugin that uses this name in IJ we are safe.

File: src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            structure.fireModified(selected);
+        }
+
+    }
+
+    @Override
+    public void update(AnActionEvent e) {
+        if (Constants.TOOLBAR_PLACE.equals(e.getPlace())) {
+            e.getPresentation().setVisible(true);
+            Tree t = getTree(e);
```
Comment (lstocchi) [inline src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java:51]: i'm facing some errors inside the `update` method.
Below a couple of them. They are always the same. Problem with casting

image

image

**Review by lstocchi — CHANGES_REQUESTED** [review]
I still see the toolbar disappear. It would be great if this could be stuck to the top.
refresh_disappear

Other thing (personal preference), could we move the refresh icon to the right of the toolbar? Like the icons above?

File: src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java (hunk 3 above)
Comment (evidolob) [reply src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java:51]: This should be fixed with https://github.com/redhat-developer/intellij-common/pull/45 maybe you need to refresh dependency

Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java:51]: Ah sorry, yes, my fault then. I'll try it with the new common.

Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/actions/RefreshAction.java:51]: I confirm it works fine with the new common. Sorry for the wrong report :+1:

**Comment by evidolob** [comment]
> Other thing (personal preference), could we move the refresh icon to the right of the toolbar? Like the icons above?

I know how to put action on right, but for some reasons, IJ put them only if there are some actions on left side

**Comment by lstocchi** [comment]
Other issue we could have is that by switching to `SimpleToolWindowPanel` we lose the scrollbar in the tree. If a user has a crowdy cluster it could be a problem. ... Don't you see the bottom of your tree? We don't care :sweat_smile: lol
Not sure if there is another panel with these features or we need to create a customized one so to have the toolbar to the top (allowing icons to the right) and the scrollbar. @jeffmaury do you know if there is such a thing?

**Comment by evidolob** [comment]
@lstocchi That was may mistake, I'm still very new on Java UI

**Comment by lstocchi** [comment]
@evidolob don't worry, i'm also not an expert of this stuff. I was joking as I hate IJ a bit. It must always complicate everything. I miss the old days with javascript/css :laughing:

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**